integration-libs/s4om/components/order-detail-attachments/order-detail-attachments.component.ts

```diff
@@ -0,0 +1,65 @@
+/*
+ * SPDX-FileCopyrightText: 2025 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  ChangeDetectionStrategy,
+  Component,
+  DestroyRef,
+  ElementRef,
+  inject,
+  OnInit,
+  ViewChild,
+  ViewContainerRef,
+} from '@angular/core';
+import { OrderDetailsService } from '@spartacus/order/components';
+import { LaunchDialogService, LAUNCH_CALLER } from '@spartacus/storefront';
+import { Observable } from 'rxjs';
+import { take } from 'rxjs/operators';
+import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
+import { Order } from '@spartacus/order/root';
+
+@Component({
+  selector: 'cx-order-details-attachments',
+  templateUrl: './order-detail-attachments.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  standalone: false,
+})
+export class OrderDetailAttachmentsComponent implements OnInit {
+  @ViewChild('element') element: ElementRef;
+  protected destroyRef = inject(DestroyRef);
+  order$: Observable<Order>;
+
+  constructor(
+    protected orderDetailsService: OrderDetailsService,
```

Comment:
MAJOR: use inject() instead of constructor for dependency injection
https://wiki.one.int.sap/wiki/display/spar/Coding+Guidelines#CodingGuidelines-Useinject()functioninsteadofconstructorforDependencyInjection